### DISCUSSION TITLE
Throttling not working for parallel request.

### DISCUSSION THREAD

**Discussion by mamidinilesh** [kickstarter/rack-attack#537] [Questions (Q&A)]
I have implemented throttling, on host
```
class Rack::Attack
  throttle('limit Developer APIs per subdomain', limit: 10, period: 10) do |req|
    if req.path.include?("/api/v2")
      req.host
    end
  end
end
```

Throttling works fine and return 429 when i hit sequentially 15 request in 6-7 secs.  
But when i hit 50 request in 1 seconds, more than 30 requests are getting processed instead of blocking after 10 request.

**Comment by nileshmamidi** [comment]
@grzuy can provide solution for this.

**Comment by grzuy** [comment]
Hi @mamidinilesh ,

Can you please provide the following details to help me better understand the issue?

Is that on production?
Is it for a Rails app?
Are is a multi threaded environment?
What is `Rack::Attack.cache.store` set to?
Which version of rack-attack are you using?

Thanks.

**Comment by nileshmamidi** [comment]
> Hi @mamidinilesh ,
> 
> Can you please provide the following details to help me better understand the issue?
> 
> Is that on production?
> Is it for a Rails app?
> Are is a multi threaded environment?
> What is `Rack::Attack.cache.store` set to?
> Which version of rack-attack are you using?
> 
> Thanks.

Hi @grzuy ,

Is that on production? - yes
Is it for a Rails app? - yes. Rails 5.2
Are is a multi threaded environment? -  Yes
What is Rack::Attack.cache.store set to? - Memcache
Which version of rack-attack are you using? rack-attack (6.2.2)

**Comment by nileshmamidi** [comment]
> > Hi @mamidinilesh ,
> > Can you please provide the following details to help me better understand the issue?
> > Is that on production?
> > Is it for a Rails app?
> > Are is a multi threaded environment?
> > What is `Rack::Attack.cache.store` set to?
> > Which version of rack-attack are you using?
> > Thanks.
> 
> Hi @grzuy ,
> 
> Is that on production? - yes
> Is it for a Rails app? - yes. Rails 5.2
> Are is a multi threaded environment? - Yes
> What is Rack::Attack.cache.store set to? - Memcache
> Which version of rack-attack are you using? rack-attack (6.2.2)

@grzuy did you found anything??

**Reply by grzuy** [reply]
Are you able to reproduce this issue in a clean new rails app?
